src/usePreferredColorScheme.ts

```diff
@@ -0,0 +1,28 @@
+import useMedia from './useMedia';
+
+/**
+ * Tracks color scheme preference.
+ *
+ * @returns {'light' | 'dark' | 'no-preference' | null} preferred color scheme
+ *
+ * @example
+ * function Example() {
+ *   const const preferredColorScheme = usePreferredColorScheme();
+ *   const isDarkMode = usePreferredColorScheme() === 'dark';
+ *   // ...
+ * }
+ */
+export default function usePreferredColorScheme():
+  | 'dark'
+  | 'light'
+  | 'no-preference'
+  | null {
+  const isLight = useMedia('(prefers-color-scheme: light)');
+  const isDark = useMedia('(prefers-color-scheme: dark)');
+  const noPreference = useMedia('(prefers-color-scheme: no-preference)');
```

Comment:
I don’t think we should distinguish between `’no-preference’` and `null`. This line should be removed in its entirely and at the bottom of the function, just return `’no-preference’` if none of the media queries match.